tracer/src/Datadog.Trace/AgentProcessManager.cs

```diff
@@ -40,13 +41,9 @@ internal class TracingProcessManager
             ProcessArguments = EnvironmentHelpers.GetEnvironmentVariable(ConfigurationKeys.DogStatsDArgs),
         };
 
-        private static readonly List<ProcessMetadata> Processes = new List<ProcessMetadata>()
-        {
-            TraceAgentMetadata,
-            DogStatsDMetadata
-        };
+        private static readonly List<ProcessMetadata> Processes = new List<ProcessMetadata>(2);
```

Comment:
Is there any way this method can be called more than once? If so we should think about concurrent modification of this list. I assume not, but maybe worth checking

Reply:
I don't see how honestly. I haven't seen the use case in my tests either.